aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -187,11 +187,11 @@ __global__ void nll_loss_forward_reduce_cuda_kernel_1d(
     index_t* target,
     scalar_t* weights,
     bool size_average,
-    int n_classes,
+    int64_t n_classes,
     int64_t ignore_index) {
   CUDA_KERNEL_ASSERT(threadIdx.x == 0 && threadIdx.y == 0 && threadIdx.z == 0);
 
-  int t = static_cast<int>(*target);
+  int64_t t = static_cast<int64_t>(*target);
   if (t != static_cast<int>(ignore_index)) {
```

Comment:
I would grep for `int[^36]` in these files and audit all the uses of `int`s in these files to make sure that's consistent.